wrangler-core/src/main/java/io/cdap/directives/row/RecordRegexFilter.java

```diff
@@ -107,10 +109,16 @@ public void destroy() {
           if (matchPattern((row.getValue(idx)).toString())) {
             continue;
           }
+        } else if (object == null) {
+          throw new DirectiveExecutionException(
+            String.format("Error encountered while executing '%s' : Column '%s' has null value. " +
+                            "It should be of type 'String', 'JSONObject' or 'Number'.", NAME, column)
```

Comment:
similar comment everywhere a null check is performed.

be of type -> be a non-null